sparsesim/src/lib.rs

```diff
@@ -519,6 +568,39 @@ impl QuantumSim {
         }
     }
 
+    fn flush_ops(&mut self) {
+        if !self.op_queue.is_empty() {
+            let ops = self
+                .op_queue
+                .iter()
+                .map(|(ctls, target, op)| {
+                    let (target, ctls) = self.resolve_and_check_qubits(*target, ctls);
+                    (ctls, target, *op)
+                })
+                .collect::<Vec<_>>();
+            self.op_queue.clear();
```

Comment:
Not sure this will compile, needs testing.
```suggestion
            let ops = self
                .op_queue
                .drain(..)
                .map(|(ctls, target, op)| {
                    let (target, ctls) = self.resolve_and_check_qubits(*target, ctls);
                    (ctls, target, *op)
                })
                .collect::<Vec<_>>();
```

Reply:
The issue with using `drain` is that it holds a mutable reference to the container, which holds a mutable reference to `self` and prevents the call inside to `self.resolve_and_check_qubits` that needs an immutable borrow.